pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -218,6 +220,43 @@ func (iqr *IQR) ReadAllColumns() (map[string][]utils.CValueEnclosure, error) {
 	}
 }
 
+func (iqr *IQR) GetAllColumnNames() (map[string]struct{}, error) {
```

Comment:
FYI: https://github.com/siglens/siglens/pull/1766 also made a method like this. We should only keep one version of this method, so if that PR gets merged first you should use it from that (and also, maybe check if you find any bugs with that implementation)

Reply:
addressed in the latest commit.